### ISSUE TITLE
template for multi-panel VCS maps

### ISSUE THREAD

**Issue by gleckler1** [PCMDI/pcmdi_metrics#408]
@lee1043   JW and PG to work on template for producing 4 panel maps with VCS module.   Driven by arguments using argparse or just as a function.    The top row (two) panels will be for the mod and obs for a given variable, with a color scale centered between and just below the two plots.   The bottom row (2 plots) will be for two difference plots (MOD-OBS and MMM-OBS), with a color scale centered between and just below these (lower row) plots.    The following argument s will need to be inputs with argparse or as a function:

1) 2D model data
2) 2D obs data
3) MMM data
3) MAX/MIN color scale for variable  (maintained in a dictionary)
4) MAX/MIN dif map color scale for variable (maintained in a dictionary)
5) OUTPATH name and location of plot produced

**Comment by gleckler1** [comment]
@lee1043 branch created with sample script for producing mean climate makes 408_pgjw_vcs ... JL will likely need PG to change some permissions...

**Comment by durack1** [comment]
@gleckler1 want to tag a milestone on this one?

**Comment by gleckler1** [comment]
@durack1 no need ... its being worked on right now

**Comment by gleckler1** [comment]
@doutriaux1  @lee1043 VCS grinding to a halt....   CD your hints yesterday helped a little but our plotting is still blowing up...  any hints??

https://github.com/PCMDI/pcmdi_metrics/blob/408_pgjw_vcs/src/python/devel/mean_climate_maps/scripts/mk_clim_dif_maps-driver.py

(pmpn) bash-4.1$ python -i mk_clim_dif_maps-driver.py

pr   ACCESS1-0   djf   plotting time is  2.39182496071
pr   ACCESS1-3   djf   plotting time is  2.26412796974
pr   bcc-csm1-1   djf   plotting time is  2.79654502869
pr   bcc-csm1-1-m   djf   plotting time is  2.8208630085
pr   BNU-ESM   djf   plotting time is  3.49944114685
pr   CanCM4   djf   plotting time is  3.97322416306
pr   CanESM2   djf   plotting time is  4.36610293388
pr   CCSM4   djf   plotting time is  4.64684605598
pr   CESM1-BGC   djf   plotting time is  5.06131196022
pr   CESM1-CAM5   djf   plotting time is  5.70535993576
pr   CESM1-FASTCHEM   djf   plotting time is  6.31002402306
pr   CESM1-WACCM   djf   plotting time is  6.93908500671
pr   CMCC-CESM   djf   plotting time is  7.90010213852
pr   CMCC-CM   djf   plotting time is  8.56252789497
pr   CMCC-CMS   djf   plotting time is  9.59797382355
pr   CNRM-CM5   djf   plotting time is  10.0138869286
pr   CSIRO-Mk3-6-0   djf   plotting time is  11.1462779045
pr   FGOALS-g2   djf   plotting time is  11.9217460155
pr   FIO-ESM   djf   plotting time is  13.1639180183
pr   GFDL-CM2p1   djf   plotting time is  14.0019130707
pr   GFDL-CM3   djf   plotting time is  15.0319869518
pr   GFDL-ESM2G   djf   plotting time is  16.0745439529
pr   GFDL-ESM2M   djf   plotting time is  17.1069629192
pr   GISS-E2-H-CC   djf   plotting time is  18.3206009865
pr   GISS-E2-H   djf   plotting time is  19.4769649506
.....
pr   CCSM4   son   plotting time is  879.395362139
pr   CESM1-BGC   son   plotting time is  744.764668941
pr   CESM1-CAM5   son   plotting time is  840.355547905
pr   CESM1-FASTCHEM   son   plotting time is  827.120736122
pr   CESM1-WACCM   son   plotting time is  837.914391994
pr   CMCC-CESM   son   plotting time is  802.292319059
pr   CMCC-CM   son   plotting time is  778.060385227
pr   CMCC-CMS   son   plotting time is  795.812077999
pr   CNRM-CM5   son   plotting time is  820.447818041
pr   CSIRO-Mk3-6-0   son   plotting time is  771.101978064
pr   FGOALS-g2   son   plotting time is  759.900727987
pr   FIO-ESM   son   plotting time is  813.396212101
pr   GFDL-CM2p1   son   plotting time is  853.033426046
pr   GFDL-CM3   son   plotting time is  925.756278992
pr   GFDL-ESM2G   son   plotting time is  905.999803782
pr   GFDL-ESM2M   son   plotting time is  912.407155991
pr   GISS-E2-H-CC   son   plotting time is  969.787393093
pr   GISS-E2-H   son   plotting time is  950.242879152
pr   GISS-E2-R-CC   son   plotting time is  914.742507935

**Comment by doutriaux1** [comment]
That line is horrible!!!!!

        execfile('./plot_map_4panel.py')


use imports. I’ll clean this up! There’s so many things worng with this… (hard coded path, not using argparse etc….)

C.

**Comment by durack1** [comment]
@gleckler1 @doutriaux1 @lee1043 @chaosphere2112 this issue is likely related to #1424 which last time I checked was still a big problem. Alongside your time getting longer and longer, the memory usage is also like blowing up too

**Comment by doutriaux1** [comment]
@durack1 while this might be true, with some clean up I was able to get a constant speed on pngs. A LOT of unnecessary objects were created.

**Comment by gleckler1** [comment]
@doutriaux1 @zshaheen trying to test CD improvements but can get everything recognized.   Starting with a new PMP env, "from pcmdi_metrics.mean_climate_maps import plot_4panel" is not recognized which is fixed by running setup.py with --enable-devel.   But then from "pcmdi_metrics.pcmdi import pmp_parser" does not work so I follow ZS instructions ... https://github.com/PCMDI/pcmdi_metrics/wiki/PMPParser and then of course loose mean_climate_maps.    I've just blown an hour trying to figure this out while simultaneously Jerome has too trying to get the json class back.    Please help us get these installations conflicts resolved.

**Comment by doutriaux1** [comment]
@gleckler1 I told you in email you need to add the cdp package:
```
conda install cdp -c uvcdat
````

**Comment by durack1** [comment]
@doutriaux1 can this be added as a conda dependency to avoid problems again in the future?

**Comment by doutriaux1** [comment]
yes it is already added. But all conda packages have been built before this. Remember we want to push a new version soon.

**Comment by durack1** [comment]
@doutriaux1 yes I think tagging a new release ASAP would be a good idea to close out this installation issue and get things back to a smoother install process

**Comment by zshaheen** [comment]
Until we have a cron job for nightly releases, we might think about adding in a section to the wiki for getting the latest version of pmp. Even if that just entails running `python setup.py install` and `conda install cdp -c uvcdat`.
